src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,149 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
```

Comment:
You can try not creating another method for checking indexes for add, but  using if clause.
```suggestion
        if (index != size) {
            rangeCheck(index);
        }   
```